kat-kan/kawqa-gad-playwright
# [ISSUE#48] add https status code enum

## Resolves #48 

### Scope of changes
We decided to introduce an enum that contains a list of the status codes currently in use. 
This will improve the overall project maintainability, and at the same time save us from code repetition.

- enums directory created in `src\api\enums`
- enum file created: `src\api\enums\api-status-code.enum.ts` with 5 status codes: 200, 400, 401, 404, 422


### How to use it in tests
- import file via:
```
import { HttpStatusCode } from "@_src_api/enums/api-status-code.enum";
```

-  when writing assertion use enum name and enum's member name, e.g.: 
```
expect(code).toBe(HttpStatusCode.Unauthorized401);
```

- when writing an assertion that uses status code that is not a member of the enum, please adjust the file and add it to the enum by following convention: `{name}{code} = {code}`
You can check the name using this [gist](https://gist.github.com/RWOverdijk/6cef816cfdf5722228e01cc05fd4b094) 

### PR steps completed:
- [x] The PR is associated with the relevant ISSUE
- [x] All new and existing tests have PASS status
- [x] Commit messages follow the `kawqa-gad-playwright` project's [git commit message format](https://github.com/kat-kan/kawqa-gad-playwright/CONTRIBUTING.md)


### Additional changes - not related to the initial PR
- Husky pre-commit: add feat and fix to branch validation regex `\.husky\pre-commit`
